### PR TITLE
Yf fix il to bed sorting

### PR DESCRIPTION
**Pull request by yfarjoun**
IntervalListToBed used Interval::compare for sorting the output but this uses a lexicographic order on the contig names...this was defintly not intended. the tests passed since the dictionaries were also in that order...

- using the correct sorting
- add better tests
- use a data provider in the tests.

### CODE REVIEW

File: testdata/picard/util/interval_list_to_bed_test.bed [hunk 1]
```diff
@@ -2,12 +2,12 @@ chr1	0	50	target_at_chrom_start	333	+
...
 chr5	4999	5150	overlapping_1	333	+
 chr5	5099	5250	overlapping_2	333	-
 chr5	5199	5350	overlapping_3	333	-
 chr5	5299	5450	overlapping_4	333	+
 chr5	9999	10000	one_base_feature	333	+
 chr5	10000	10000	zero_base_feature	333	+
-chr5	999999	1000000	target_at_chrom_end	333	+
\ No newline at end of file
+chr5	999999	1000000	target_at_chrom_end	333	+
+chr4	399	440	target_4	333	+
```
Comment (nh13) [inline testdata/picard/util/interval_list_to_bed_test.bed:13]: clever

File: src/test/java/picard/util/IntervalListToBedTest.java [hunk 2]
```diff
@@ -14,57 +15,41 @@
...
-        final IntervalListToBed program = new IntervalListToBed();
-        final File tmp = File.createTempFile("interval_list_to_bed_test_output", ".bed");
-        tmp.deleteOnExit();
-
-        final String[] args = {
-                "INPUT=" + INTERVAL_LIST.getAbsolutePath(),
-                "OUTPUT=" + tmp.getAbsolutePath(),
-                "SCORE=333"
+    private final File BED_FILE = new File(TEST_DATA_DIR, "interval_list_to_bed_test.bed");
+    private final File NO_SORT_BED_FILE = new File(TEST_DATA_DIR, "no_sort_interval_list_to_bed_test.bed");
```
Comment (nh13) [inline src/test/java/picard/util/IntervalListToBedTest.java:19]: Any chance you can write 1-2 lines of comments about the properties of these BED files (ex chr5/chr4 are out of karyotype order)?  That way we don't have to look at the test files to know (but can look to verify).

File: src/test/java/picard/util/IntervalListToBedTest.java [hunk 3]
```diff
@@ -14,57 +15,41 @@
...
+
+    @DataProvider()
+    Object[][] testConvertILToBedData() {
+        return new Object[][]{
+                {INTERVAL_LIST, BED_FILE, true},
+                {INTERVAL_LIST, NO_SORT_BED_FILE, false},
+                {UNSORTED_INTERVAL_LIST, BED_FILE, true},
```
Comment (nh13) [inline src/test/java/picard/util/IntervalListToBedTest.java:26]: some comments on what's being tested would be nice.  Even "out-of-order gets put into karyotype order" or "seq dict is out of karyotype order, so sorted interval list will be too"

**Review by nh13 — COMMENTED** [review]
I am not a reviewer, but I added some comments anyhow, because.  I think there could be some better comments about the tests and their motivation, to help our future selves.

**Comment by yfarjoun** [comment]
@nh13 better now?
@tfenne you complained about this in private...would you like to review?

**Comment by gbggrant** [comment]
@nh13 @tfenne do either of you want to re-review @yfarjoun 's updates here?

**Comment by nh13** [comment]
LGTM!

**Comment by yfarjoun** [comment]
@gbggrant needs a "approving review"....

**Review by gbggrant — APPROVED** [review]

**Comment by gbggrant** [comment]
All set @yfarjoun

**Pull request merged**